utilities/validations.py

```diff
@@ -66,3 +70,14 @@ def validate_timezone_field(**kwargs):
     time_zone = kwargs['time_zone']
     if time_zone not in timezones:
         raise AttributeError("Not a valid time zone")
+
+
+def validate_parent_node_id(parent_id):
+    """
+    Function to validate Parent node ID
+    when creating child node
+    """
+    if parent_id:
```

Comment:
What happens if the `node_id` is not provided?

Reply:
If a node_id(parent_id) is not provided, that means a new root has been created hence a new tree

Reply:
so what am doing here is trying to validate the parent_id once provided. The validation is about preventing the creation of a node with a parent_id that doesn't exist.

Reply:
One has to always provide a parent_node when creating a child for that parent

Reply:
Hey @Bonifase , I am not validating this in the schema. This check is necessary because if I don't include it then one will just create any node and provide a parent_id that does not exist. Even though the parent_id field of that child won't be populated, I think it is necessary to warn the user that the parent_id he or she is trying to use does not exist.

Reply:
> Hey @Bonifase , I am not validating this in the schema. This check is necessary because if I don't include it then one will just create any node and provide a parent_id that does not exist. Even though the parent_id field of that child won't be populated, I think it is necessary to warn the user that the parent_id he or she is trying to use does not exist.

My concern is why use the `if parent_id:` statement yet it has to be provided always. Why can't you just do the query in line 79 without line 78?

Reply:
Awesome work @nakatuddesuzan this `if parent_id:` is redundant thus doing away with it, in this case, is a good option.

Reply:
@Bonifase and @dnuwa  , That line is necessary `parent_id` is not always providedfor examplee when creating a root not(That is when starting a new tree). Parent_ids are only provided when creating child nodes. So that `if statement` is checking when for when the parent_id is provided

Reply:
For clarity here, because the root is created without a parent_id,  if i remove the check ('if parent_id'). I won't be able to create that root as it will be checking for a parent_id and yet it is not provided. You can try this out your end. Remove the if statement and run this mutation, you will prove what am taking about and understand it better
```
mutation{
  createNode(name:"root"){
    node{
      name
      level
      left
      right
      parentId
    }
    
  }
}
```

Reply:
Hello @Bonifase , Thanks for the pointers I have more made it more readable like advised